src/filters/filter_lua_script.cpp

```diff
@@ -0,0 +1,382 @@
+                throw filter_config_exception("Unable to link input of lua filter: channel mapping does not contain channel '" + names_in_sixteen_bit.at(i) + "' of type 'uint16_t'.");
+            }
+            in_sixteen_bit.at(i) = input_channels.sixteen_bit_channels.at(names_in_sixteen_bit.at(i));
+        }
+        for (size_t i = 0; i < names_in_float.size(); i++) {
+            if(!input_channels.float_channels.contains(names_in_float.at(i))) {
+                throw filter_config_exception("Unable to link input of lua filter: channel mapping does not contain channel '" + names_in_float.at(i) + "' of type 'double'.");
+            }
+            in_float.at(i) = input_channels.float_channels.at(names_in_float.at(i));
+        }
+        for (size_t i = 0; i < names_in_color.size(); i++) {
+            if(!input_channels.color_channels.contains(names_in_color.at(i))) {
+                throw filter_config_exception("Unable to link input of lua filter: channel mapping does not contain channel '" + names_in_color.at(i) + "' of type 'pixel'.");
+            }
+            in_color.at(i) = input_channels.color_channels.at(names_in_color.at(i));
+            // Todo delete in_color;
+            lua[names_in_color.at(i)] = std::ref(input_channels.color_channels.at(names_in_color.at(i)));
+        }
+
+        if (!initial_parameters.contains("script")) {
+            throw filter_config_exception("lua filter: unable to setup the script");
+        }
+
+//        if (!initial_parameters.contains("patching")) {
+//            throw filter_config_exception("lua filter: unable to setup the patching");
+//        }
+
+        // initial_parameters.at("patching") divide
+        // for each fixture
+        //      fix_univ
+        //      fix_name
+        //      fix_ch_size
+        //      fixture local = fixture(fix_univ, fix_name, fix_ch_size);
+        //      ? dynamisch verschieden viele Argumente übergeben?
+//              lua.new_usertype<fixture>("Fixture", "pan", &fixture.channel_values.at(0), "s", &dmxfish::dmx::pixel::saturation, "i", &dmxfish::dmx::pixel::iluminance);
+        //      fixtures.append(local);
+
+
+
+        lua.open_libraries(sol::lib::base, sol::lib::package);
+        lua.open_libraries(sol::lib::math, sol::lib::table, sol::lib::string);
+        lua.set_function("update", []() {
+        });
+        lua.set_function("scene_activated", []() {
+        });
+
+//        for(fixture fix: fixtures){
+//            lua.create_named_table(fix.name, "universe", fix.universe,
+//                                   "first_channel", fix.first_channel);
+//        }
+        lua.create_named_table("output");
+
+        lua.set_function( "hsi_to_rgb", sol::overload(
+                hsi_to_rgb_color,
+                hsi_to_rgb_table
+        ) );
+
+        lua.set_function( "hsi_to_rgbw", sol::overload(
+                hsi_to_rgbw_color,
+                hsi_to_rgbw_table
+        ) );
```

Comment:
As we should out source the methods of the API, so should we do for context creation as we might need this behaviour in multiple places as well.